### PR TITLE
Fixes for error: invalid byte sequence in UTF-8 (ArgumentError).

### PR DESCRIPTION
**Pull request by mfragkoulis**
Fixes issue #458

### CODE REVIEW

**Comment by rodjek** [comment]
If you modify `content` at the start of `PuppetLint::Checks#load_data`, you only have to change the encoding once which'll be more efficient.

File: lib/puppet-lint/plugins/check_whitespace.rb [hunk 1]
```diff
@@ -112,6 +112,9 @@ def check
         if token.type == :FARROW
           (level_tokens[indent_depth_idx] ||= []) << token
           prev_indent_token = resource_tokens[0..idx].rindex { |t| t.type == :INDENT }
+          if not prev_indent_token
+            next
+          end
```
Comment (rodjek) [inline lib/puppet-lint/plugins/check_whitespace.rb]: What's the purpose of this change?

**Comment by ahpook** [comment]
@mfragkoulis this is failing tests and we, like @rodjek are not clear on what you're trying to do here.

**Comment by mfragkoulis** [comment]
Some months back I used Puppet-lint on a research project, but it was crashing when executed on some projects with prev_indent_token having a NULL value.
I tracked down the error and provided a quick and dirty fix.
Then I decided to contribute back the change with a pull request in case the bug was as simple as that.
With the tests failing, unfortunately I didn't have time to dive deeper to the source of the problem.
My computing environment at the time was a Mac OS X Yosemite running Ruby 1.9.3 I think.
If you decide to investigate this and want me to test any patches you produce feel free to drop me a line.

FYI: I wasn't following up on the pull request, I got an email with the mentioning of my name two days ago.

**Comment by rnelson0** [comment]
MongoDB started popping this error in builds after 2.0.0 was released, for example https://travis-ci.org/puppetlabs/puppetlabs-mongodb/builds/141083547. I'm not sure what file is causing it but I'm sure we can track that down.

**Comment by rnelson0** [comment]
Nevermind. This instance was more related to not adding `vendor/` to the exclude paths, and a selinux policy which matches the `*.pp` pattern. @mfragkoulis I don't know if there's any chance that you think that may have been part of the cause? Your fix stopped the alarm, but scanning a selinux policy doesn't add much value :)
```
$ file ./vendor/ruby/gems/puppet-4.5.2/spec/unit/provider/selmodule-example.pp
./vendor/ruby/gems/puppet-4.5.2/spec/unit/provider/selmodule-example.pp: SE Linux modular policy version 1, 2 sections, mod version 8, MLS, module name nagios\005
$ be puppet-lint ./vendor/ruby/gems/puppet-4.5.2/spec/unit/provider/selmodule-example.pp
/home/rnelson0/git/mongodb/vendor/ruby/gems/puppet-lint-2.0.0/lib/puppet-lint/checks.rb:23:in `split': invalid byte sequence in UTF-8 (ArgumentError)
        from /home/rnelson0/git/mongodb/vendor/ruby/gems/puppet-lint-2.0.0/lib/puppet-lint/checks.rb:23:in `load_data'
        from /home/rnelson0/git/mongodb/vendor/ruby/gems/puppet-lint-2.0.0/lib/puppet-lint/checks.rb:53:in `run'
        from /home/rnelson0/git/mongodb/vendor/ruby/gems/puppet-lint-2.0.0/lib/puppet-lint.rb:174:in `run'
        from /home/rnelson0/git/mongodb/vendor/ruby/gems/puppet-lint-2.0.0/lib/puppet-lint/bin.rb:59:in `block in run'
        from /home/rnelson0/git/mongodb/vendor/ruby/gems/puppet-lint-2.0.0/lib/puppet-lint/bin.rb:56:in `each'
        from /home/rnelson0/git/mongodb/vendor/ruby/gems/puppet-lint-2.0.0/lib/puppet-lint/bin.rb:56:in `run'
        from /home/rnelson0/git/mongodb/vendor/ruby/gems/puppet-lint-2.0.0/bin/puppet-lint:7:in `<top (required)>'
        from /home/rnelson0/git/mongodb/vendor/ruby/bin/puppet-lint:23:in `load'
        from /home/rnelson0/git/mongodb/vendor/ruby/bin/puppet-lint:23:in `<main>'
```

**Comment by mfragkoulis** [comment]
Yes, that could be the cause. I was recursively and greedily analysing multiple projects.

**Comment by rnelson0** [comment]
@mfragkoulis I'm going to close this in preference of a safer failure as proposed in #458.
